src/pose_publisher.cpp

```diff
@@ -1,85 +1,130 @@
+
+using namespace std::chrono_literals;
+
+class PosePubNode : public rclcpp::Node
+{
+public:
+    PosePubNode() : Node("node_pose_publisher")
+    {
+        RCLCPP_INFO(get_logger(),  "Connection established");
+
+        // Read parameters from the parameter server
+        RCLCPP_DEBUG(get_logger(), "Reading parameters from server");
+        read_parameters();
+
+        // Create ROS2 tf listener
+        tf_buffer_ =
+            std::make_unique<tf2_ros::Buffer>(this->get_clock());
+        tf_listener_ =
+            std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+
+        // Create pose_publisher publisher
+        pose_publisher_ =
+            this->create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>(topic_republish, 1);        
+        RCLCPP_INFO(get_logger(),  "Pose_publisher created with topic: %s", topic_republish.c_str());
+        
+        // std::chrono::seconds timer_period(1);
+        std::chrono::milliseconds timer_period{500};
+
+        // Call timer_callback function every timer_period
+        timer_ = this->create_wall_timer(
+            timer_period, std::bind(&PosePubNode::timer_callback, this));
+    }
+
+    void read_parameters()
+    {
+        // Declare and acquire parameters
+        declare_parameter<std::string>("robot", tf_prefix);
+        declare_parameter<std::string>("map", map_frame);
+        declare_parameter<std::string>("base_link", base_frame);
+        declare_parameter<std::string>("pose", topic_republish);
+
+        get_parameter("robot", tf_prefix);
+        get_parameter("map", map_frame);
+        get_parameter("base_link", base_frame);
+        get_parameter("pose", topic_republish);
     }
 
-    if(tf_ok) {
-      geometry_msgs::PoseWithCovarianceStamped pose_stamped;
-      pose_stamped.header.stamp = ros::Time::now();
-      pose_stamped.header.frame_id = tf_prefix+"/"+map_frame;
 
-      pose_stamped.pose.pose.position.x = transform.getOrigin().getX();
-      pose_stamped.pose.pose.position.y = transform.getOrigin().getY();
-      pose_stamped.pose.pose.position.z = transform.getOrigin().getZ();
+private:
+    void timer_callback()
+    {
+        RCLCPP_INFO(get_logger(),  "In timer_callback");
+
+        std::string fromFrameRel = (tf_prefix + "/" + base_frame).c_str();
+        std::string toFrameRel = (tf_prefix + "/" + map_frame).c_str();
+
+        // std::string fromFrameRel = (tf_prefix + "/" + map_frame).c_str();
+        // std::string toFrameRel = (tf_prefix + "/" + base_frame).c_str();
+
+        RCLCPP_INFO(get_logger(),  "fromFrameRel topic (target_frame): %s", fromFrameRel.c_str());
+        RCLCPP_INFO(get_logger(),  "toFrameRel topic (source_frame): %s",toFrameRel.c_str());
+
+        geometry_msgs::msg::TransformStamped tf_transform;
+        bool tf_ok = true;
+
+        try {
+            tf_transform = tf_buffer_->lookupTransform(
+                toFrameRel, fromFrameRel,
+                tf2::TimePointZero);
```

Comment:
If I remember well, this use latest available transform from buffer. Should be used current timestamp of wall timer?

https://github.com/ros2/geometry2/blob/ad72a82961e82470d2c9848840dd6e8eee44b127/tf2_ros/include/tf2_ros/buffer.h#L137